embassy-rp/src/clocks.rs

```diff
@@ -1,11 +1,168 @@
+use embassy_hal_common::{into_ref, PeripheralRef};
 use pac::clocks::vals::*;
 
-use crate::{pac, reset};
+use crate::{pac, reset, Peripheral};
 
-const XOSC_MHZ: u32 = 12;
+// TODO fix terrible use of global here
+static mut XIN_HZ: u32 = 0;
+
+pub struct ClockConfig {
```

Comment:
- fields should be public, otherwise you can't change them
- add `#[non_exhaustive]`, so that we can add fields in the future without it being a breaking change.